internal/buildsign/pod/signer.go

```diff
@@ -0,0 +1,266 @@
+package pod
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"os"
+	"text/template"
+
+	"github.com/mitchellh/hashstructure/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/kubernetes-sigs/kernel-module-management/internal/api"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/module"
+	buildsignpod "github.com/kubernetes-sigs/kernel-module-management/internal/pod"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+)
+
+type TemplateData struct {
+	FilesToSign   []string
+	SignImage     string
+	UnsignedImage string
+}
+
+//go:embed templates
+var templateFS embed.FS
+
+var tmpl = template.Must(
+	template.ParseFS(templateFS, "templates/Dockerfile.gotmpl"),
+)
+
+//go:generate mockgen -source=signer.go -package=pod -destination=mock_signer.go
+
+type Signer interface {
+	MakePodTemplate(
```

Comment:
I am wondering why don't we move that method to the pod package in the BuildSignPodManager interface instead of a separate interface in a different package to also manager build/sign pods?

Reply:
We have agreed that `Maker` and `Signer` will be removed in a later PR and their method will be merge into the `BuildSignPodManager` interface instead. Some refactor is needed so it will be handled in a separate PR.